src/main/webapp/app/shared/metis/metis.service.ts

```
@@ -668,13 +690,28 @@ export class MetisService implements OnDestroy {
+                    const currentPinnedPosts = this.pinnedPosts$.getValue();
+                    const alreadyPinned = currentPinnedPosts.some((pinnedPost) => pinnedPost.id === postDTO.post.id);
+                    if (!alreadyPinned) {
+                        this.pinnedPosts$.next([postDTO.post, ...currentPinnedPosts]);
+                    }
+                } else {
+                    this.removeFromPinnedPosts(postDTO.post.id!);
+                }
                 this.addTags(postDTO.post.tags);
                 break;
             case MetisPostAction.DELETE:
                 const indexToDelete = this.cachedPosts.findIndex((post) => post.id === postDTO.post.id);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)